sdk/identity/Azure.Identity/tests/FileContentsCacheTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class FileContentsCacheTests
+    {
+        private TestTempFileHandler _tempFiles = new TestTempFileHandler();
+
+        [Test]
+        public async Task VerifyCacheAndRefresh()
+        {
+            string originalText = Guid.NewGuid().ToString();
+
+            string updatedText = Guid.NewGuid().ToString();
+
+            string filePath = _tempFiles.GetTempFilePath();
+
+            File.WriteAllText(filePath, originalText);
+
+            var fileCache = new FileContentsCache(filePath, TimeSpan.FromSeconds(3));
+
+            // assert the file text is returned
+            Assert.AreEqual(originalText, await fileCache.GetTokenFileContentsAsync(default));
+
+            File.WriteAllText(filePath, updatedText);
+
+            // assert the cached file text is still returned
+            Assert.AreEqual(originalText, await fileCache.GetTokenFileContentsAsync(default));
+
+            await Task.Delay(TimeSpan.FromSeconds(4));
```

Comment:
It would be nice if we could avoid a static wait like this, but I'm good if we defer it for now.

Reply:
Talked offline and decided to shrink the wait time down and defer changing to a different mechanism to later if necessary.